### PR TITLE
Add custom method for generating an eslintrc config

### PR DESCRIPTION
**Pull request by eliperelman**
- [x] Generate an ESLint configuration usable by `.eslintrc.js`
- [x] Document a recipe for creating the eslintrc file through the API:

```js
// .eslintrc.js
const Neutrino = require('neutrino');
const pkg = require('./package.json');
const api = new Neutrino(pkg.config.presets);

module.exports = api.custom.eslintrc();
```

### CODE REVIEW

File: packages/neutrino-lint-base/src/index.js [hunk 1]
```diff
@@ -37,4 +40,27 @@ module.exports = ({ config }) => {
...
   config.resolveLoader.modules.add(MODULES);
+
+  neutrino.custom.eslintrc = () => {
+    const options = clone(config.module.rule('lint').loaders.get('eslint').options);
+
+    options.extends = options.baseConfig.extends;
+    options.useEslintrc = true;
+    options.env = options.envs.reduce((env, key) => {
+      env[key] = true;
+      return env;
```
Comment (helfi92) [inline packages/neutrino-lint-base/src/index.js]: @eliperelman Out of curiosity, in what scenario do you not prefer adding an empty line between an assignment and the return value.

Comment (eliperelman) [inline packages/neutrino-lint-base/src/index.js]: I usually always do it, I'll change

File: packages/neutrino-lint-base/src/index.js [hunk 2]
```diff
@@ -37,4 +40,27 @@ module.exports = ({ config }) => {
...
+    options.globals = options.globals.reduce((globals, key) => {
+      globals[key] = true;
+      return globals;
+    }, {});
+
+    delete options.envs;
+    delete options.baseConfig;
+    delete options.failOnError;
+    delete options.emitWarning;
+    delete options.emitError;
```
Comment (helfi92) [inline packages/neutrino-lint-base/src/index.js]: Thoughts about this instead?
```javascript
['envs', 'baseConfig', 'failOnError', 'emitWarning', 'emitError']
  .forEach(prop => delete options[prop]);
```

File: packages/neutrino-lint-base/src/index.js [hunk 3]
```diff
@@ -37,4 +40,27 @@ module.exports = ({ config }) => {
...
+
+    options.extends = options.baseConfig.extends;
+    options.useEslintrc = true;
+    options.env = options.envs.reduce((env, key) => {
+      env[key] = true;
+      return env;
+    }, {});
+    options.globals = options.globals.reduce((globals, key) => {
+      globals[key] = true;
+      return globals;
```
Comment (helfi92) [inline packages/neutrino-lint-base/src/index.js]: Empty line between assignment and return value

**Pull request merged**